src/main/java/de/tum/cit/ase/ares/api/architecture/java/CallGraphBuilderUtils.java

```diff
@@ -0,0 +1,155 @@
+        }
+    }
+
+    /**
+     * Get the immediate subclasses of the given type name.
+     *
+     * @param typeName The type name of the class to get the immediate subclasses.
+     * @return The immediate subclasses of the given type name.
+     */
+    public static Set<JavaClass> getImmediateSubclasses(String typeName) {
+        TypeReference reference = TypeReference.find(ClassLoaderReference.Application, convertTypeName(typeName));
+        if (reference == null) {
+            return Collections.emptySet();
+        }
+        IClass clazz = classHierarchy.lookupClass(reference);
+        if (clazz == null) {
+            return Collections.emptySet();
+        }
+        return classHierarchy
+                .getImmediateSubclasses(clazz)
+                .stream()
+                .map(IClass::getName)
+                .map(Object::toString)
+                .map(CallGraphBuilderUtils::tryResolve)
+                .filter(Optional::isPresent)
+                .map(Optional::get).collect(Collectors.toSet());
+    }
+
+    /**
+     * Convert the type name to the format that can be used in the class file.
+     *
+     * @param typeName The type name to convert.
+     * @return The converted type name.
+     */
+    public static String convertTypeName(String typeName) {
+        if (typeName == null || typeName.isEmpty()) {
+            throw new IllegalArgumentException("Type name cannot be null or empty");
+        }
+        return "L" + typeName.replace('.', '/');
+    }
+
+    /**
+     * Build a call graph from a class path and the passed in predicate
+     */
+    public static CallGraph buildCallGraph(String classPathToAnalyze) {
+        try {
+            // Create a list to store entry points
+            List<DefaultEntrypoint> customEntryPoints = ReachabilityChecker.getEntryPointsFromStudentSubmission(classPathToAnalyze, classHierarchy);
+
+            // Create AnalysisOptions for call graph
+            AnalysisOptions options = new AnalysisOptions(scope, customEntryPoints);
+            options.setTraceStringConstants(false);
+            options.setHandleZeroLengthArray(false);
+            options.setReflectionOptions(AnalysisOptions.ReflectionOptions.NONE);
+
+            // Create call graph builder (n-CFA, context-sensitive, etc.)
+            com.ibm.wala.ipa.callgraph.CallGraphBuilder<InstanceKey> builder = Util.makeZeroOneCFABuilder(Language.JAVA, options, new AnalysisCacheImpl(), classHierarchy);
+
+            // Generate the call graph
+            return builder.makeCallGraph(options, null);
+        } catch (CallGraphBuilderCancelException e) {
+            throw new SecurityException("Error building call graph", e); //$NON-NLS-1$
+        }
+    }
+
+    /**
+     * Get the content of a file from the architectural rules storage
+     */
+    public static Set<String> getForbiddenMethods(Path filePath) {
+        return new HashSet<>(List.of(FileTools.readFile(filePath).split("\r\n")));
```

Comment:
Why `"\r\n"`?

Reply:
It seems like in the txt file the methods are splitted via \r\n